src/credentials/StaticGroupDataProvider.cpp

```diff
@@ -0,0 +1,487 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "GroupDataProvider.h"
+#include <string.h>
+
+using namespace chip::DeviceLayer;
+
+namespace chip {
+namespace Credentials {
+namespace {
+
+static constexpr size_t GroupFabricsMax    = 4;
+static constexpr size_t EndpointEntriesMax = 16;
+static constexpr size_t StateEntriesMax    = 16;
+static constexpr size_t KeyEntriesMax      = 16;
```

Comment:
These should come from CHIP_CONFIG_XXX options

Reply:
I have included CHIPProjectConfig.h, and added default CHIP_CONFIG_XXX values for these.